File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        result = float("inf")
        for i in range(n):
            if words[i] == target:
                clockwise_dist = abs(i - startIndex)
                anti_clockwise_dist = n - clockwise_dist
                result = min(result, clockwise_dist, anti_clockwise_dist)
        return -1 if result == float("inf") else result

File: 2598-shortest-distance-to-target-string-in-a-circular-array/shortest-distance-to-target-string-in-a-circular-array.js

var closestTarget = function (words, target, startIndex) {
    const n = words.length
    let result = Infinity
    for (let i = 0; i < n; i++) {
        if (words[i] === target) {
            const clockwise_dist = Math.abs(i - startIndex)
            const anti_clockwise_dist = n - clockwise_dist

            result = Math.min(result, clockwise_dist, anti_clockwise_dist)
        }
    }
    return result === Infinity ? -1 : result
};